x-pack/legacy/plugins/lens/public/indexpattern_plugin/datapanel.tsx

```diff
@@ -324,148 +336,181 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
+                data-test-subj="lnsIndexPatternFieldSearch"
+                placeholder={i18n.translate('xpack.lens.indexPatterns.filterByNameLabel', {
+                  defaultMessage: 'Search for fields',
+                  description:
+                    'Search the list of fields in the index pattern for the provided text',
+                })}
+                value={localState.nameFilter}
+                onChange={e => {
+                  setLocalState({ ...localState, nameFilter: e.target.value });
+                }}
+                aria-label={i18n.translate('xpack.lens.indexPatterns.filterByNameAriaLabel', {
+                  defaultMessage: 'Search fields',
+                })}
+              />
+            </EuiFormControlLayout>
+          </div>
+          <div className="lnsInnerIndexPatternDataPanel__filtersWrapper">
+            <EuiPopover
+              id="dataPanelTypeFilter"
+              panelClassName="euiFilterGroup__popoverPanel"
+              panelPaddingSize="none"
+              anchorPosition="downLeft"
+              display="block"
+              isOpen={localState.isTypeFilterOpen}
+              closePopover={() => setLocalState(s => ({ ...localState, isTypeFilterOpen: false }))}
+              button={
+                <EuiFacetButton
+                  data-test-subj="lnsIndexPatternFiltersToggle"
+                  className="lnsInnerIndexPatternDataPanel__filterButton"
+                  quantity={localState.typeFilter.length}
+                  icon={<EuiIcon type="filter" />}
+                  isSelected={localState.typeFilter.length ? true : false}
+                  onClick={() => {
                     setLocalState(s => ({
                       ...s,
-                      nameFilter: '',
-                      typeFilter: [],
+                      isTypeFilterOpen: !localState.isTypeFilterOpen,
                     }));
-                  },
-                }}
-              >
-                <input
-                  className="euiFieldText euiFieldText--inGroup"
-                  data-test-subj="lnsIndexPatternFieldSearch"
-                  placeholder={i18n.translate('xpack.lens.indexPatterns.filterByNameLabel', {
-                    defaultMessage: 'Search fields',
-                    description:
-                      'Search the list of fields in the index pattern for the provided text',
-                  })}
-                  value={localState.nameFilter}
-                  onChange={e => {
-                    setLocalState({ ...localState, nameFilter: e.target.value });
                   }}
-                  aria-label={i18n.translate('xpack.lens.indexPatterns.filterByNameAriaLabel', {
-                    defaultMessage: 'Search fields',
+                >
+                  <FormattedMessage
+                    id="xpack.lens.indexPatterns.toggleFiltersPopover"
+                    defaultMessage="Fields filtered"
```

Comment:
Subjective, but I was confused by this caption - depending on the screen size there is a lot of space between the number and the caption (see also screenshot above) Maybe `Fields filtered: ` or just `Filter fields` make it clearer

Reply:
This was text directly from @AlonaNadler But you all can work with @gchaps on this language as well. I'll move it to the Lens Text Review doc.